classes/gateways/class.pmprogateway_paypalexpress.php

```diff
@@ -1068,6 +1069,37 @@ function PPHttpPost( $methodName_, $nvpStr_ ) {
 
 			return $httpParsedResponseAr;
 		}
+
+		/**
+		 * Refunds an order (only supports full amounts)
+		 *
+		 * @param bool    Status of the refund (default: false)
+		 * @param object  The Member Order Object
+		 *
+		 * @return bool   Status of the processed refund
+		 */
+		function process_refund( $success, $morder ){
+
+			$transaction_id = $morder->payment_transaction_id;
+
+			if ( $transaction_id === $morder->subscription_transaction_id ) {
+				$transaction_id = $morder->Gateway->getRealPaymentTransactionId( $morder );
+			}
+
+			$httpParsedResponseAr = $morder->Gateway->PPHttpPost( 'RefundTransaction', "&TRANSACTIONID=$transaction_id" );
+
+			if ( 'SUCCESS' === strtoupper( $httpParsedResponseAr['ACK'] ) ) {
+				$morder->updateStatus( 'refunded' );
+				return true;
+			} else {
+				//Localize this with better wording
+				$morder->notes = $morder->notes.' Attempted refund failed '.print_r( $httpParsedResponseAr, true );
```

Comment:
Should this be added with a \n\n or do we not worry about that?

Reply:
May also want to trim() the notes too.

Reply:
Trimming it in an upcoming commit, but \n\n gets rendered in the notes block. So just appending to the existing string for now.